src/http.rs

```diff
@@ -99,6 +100,42 @@ impl HttpClient {
         })
     }
 
+    /// Function `entity_timed` is like `entity` but with two optional fields `transaction_time` and `valid_time` that are of type `Option<DateTime<FixedOffset>>`.
+    pub fn entity_timed(
+        &self,
+        id: String,
+        transaction_time: Option<DateTime<FixedOffset>>,
+        valid_time: Option<DateTime<FixedOffset>>,
+    ) -> Result<Edn, CruxError> {
+        if !id.starts_with(":") {
+            return Ok(edn!({:status ":bad-request", :message "ID required", :code 400}));
+        }
+
+        let mut s = String::new();
+        s.push_str("{:eid ");
+        s.push_str(&id);
+        s.push_str("}");
+
+        let url = build_timed_url(self.uri.clone(), "entity", transaction_time, valid_time);
+
+        let resp = self
+            .client
+            .post(&url)
+            .headers(self.headers.clone())
+            .body(s)
+            .send()?
+            .text()?;
+
+        let edn_resp = edn_rs::parse_edn(&resp);
+        Ok(match edn_resp {
```

Comment:
What about instead of `Ok(match edn_resp ....)`, this:

```rust
edn_resp.map_err(|err| {
    println!(":CRUX-CLIENT POST /entity [ERROR]: {:?}", err);
    edn!({:status ":internal-server-error", :code 500})
})
```

It doesn't need the `match` and/or the `Ok` :slightly_smiling_face:

Reply:
```
expected enum `std::result::Result<_, types::error::CruxError>`
   found enum `std::result::Result<_, edn_rs::edn::Edn>
```

Reply:
In the end is the same problem as with `the error here is not associated with the process failing but with a bad-request to server, the code 400 will wrap this as the correct error. Also, we cannot build a reqwest::Error, which is lame for this reason`. We don't have a proper type to wrap this EDN and just returning a `CruxError` won't solve the `reqwest::Error` problem, which would be interesting as it is unaccessable.

Reply:
We will need to create a type to wrap this error, will do this in the examples/test PR

Reply:
Okay then, I still find very wierd that this doesn't work :thinking:

Reply:
The problem with this solution is that we are mapping the error to an EDN and we don't have that mapped in `CruxError`.  We can pair in this improvement later